code/network/multi_ping.cpp

```diff
@@ -79,21 +80,22 @@ void multi_ping_eval_pong(ping_struct *ps)
 			ps->ping_add++;
 		}
 
-		ps->ping_times[ps->ping_add] = f2fl(timer_get_fixed_seconds()) - ps->ping_start;
+		ps->ping_times[ps->ping_add] = pong_time - ps->ping_start;
+		Assertion(ps->ping_times[ps->ping_add == 0 ? MAX_PINGS - 1 : ps->ping_add - 1] > 0L, "Non-positive ping value!");
 	}
 
 	// calculate the average ping time
-	ping_sum = 0.0f;
+	ping_sum = 0L;
 	for(idx=0;idx<ps->num_pings;idx++){
 		ping_sum += ps->ping_times[idx];
-	}
-	ps->ping_avg = (int)(1000.0f * (ping_sum / (float)ps->num_pings));	
+	}// jg18 - IMO the slight loss of precision from the integer divide is better than the loss from premature double conversion
+	ps->ping_avg = (int)f2fld(1000L * ping_sum / (fix)ps->num_pings);
```

Comment:
I think doing the double convert early and then dividing by `num_pings` should be more precise. Doubles, especially in the range they are used here, are very precise and it's unlikely that converting the fix to a double would lead to any significant precision loss.